### PR TITLE
Punning support

### PR DESCRIPTION
**Pull request by SanderSpies**
Unfortunately the wrappingTest is failing due to idempotent issues with comments.

### CODE REVIEW

**Comment by yunxing** [comment]
@SanderSpies patching your diff to see what's causing the comment issue.

**Comment by yunxing** [comment]
image

Trying to print out the AST of a smiple test, looks like the location for the pattern variable "a" is missing (see highlighted area)?

**Comment by SanderSpies** [comment]
Thanks for taking a look. I was looking in this direction, but forgot about this trick.

**Comment by SanderSpies** [comment]
@yunxing Thanks again. It should work now 👍

**Comment by yunxing** [comment]
Looks good! Let's merge this after CI check.

File: docs/index.html [hunk 1]
```diff
@@ -428,7 +428,7 @@
 let thirteen = increment (double 6);
 ```
 
-####
```
Comment (jordwalke) [inline docs/index.html]: Oops! These need to stay here btw.

**Pull request merged**